{
  "comments": [
    {
      "key": {
        "uuid": "PtqwXY0s",
        "filename": "/COMMIT_MSG",
        "patchSetId": 4
      },
      "lineNbr": 7,
      "author": {
        "id": 1003873
      },
      "writtenOn": "2013-02-11T12:57:33Z",
      "side": 1,
      "message": "How to do reproduce this error?\n\nI tried:\n\n  $ mvn test\n\nboth from the gerrit-acceptance-tests project and from the parent project and couldn\u0027t get any errors.",
      "revId": "ff6e8e4a430fad80e7fb632a5424bafb28d415ff",
      "serverId": "173816e5-2b9a-37c3-8a2e-48639d4f1153",
      "unresolved": false
    },
    {
      "key": {
        "uuid": "PxhwBVRA",
        "filename": "/COMMIT_MSG",
        "patchSetId": 4
      },
      "lineNbr": 7,
      "author": {
        "id": 1010015
      },
      "writtenOn": "2013-02-11T14:53:56Z",
      "side": 1,
      "message": "I find it when I run \u0027mvn clean package\u0027 , in fact it is about \u0027mvn test\u0027\n\n $ git checkout -b retrigger b32df3609e3be35660a80af58dfa8b79e10c955b\n $ mvn test\n ...\n ...\n ...\n -------------------------------------------------------\n T E S T S\n -------------------------------------------------------\n Running com.google.gerrit.acceptance.TestAccount\n Tests run: 2, Failures: 0, Errors: 2, Skipped: 0, Time elapsed: 0.045 sec \u003c\u003c\u003c   \n FAILURE!\n\n Results :\n\n Tests in error: \n   initializationError(com.google.gerrit.acceptance.TestAccount)\n   initializationError(com.google.gerrit.acceptance.TestAccount)\n\n Tests run: 2, Failures: 0, Errors: 2, Skipped: 0\n\n [INFO] ------------------------------------------------------------------------\n [ERROR] BUILD FAILURE\n [INFO] ------------------------------------------------------------------------\n [INFO] There are test failures.\n\n Please refer to /home/xing-yi/gerrit-code/google/gerrit/gerrit-acceptance-tests/target /surefire-reports for the individual test results.\n [INFO] ------------------------------------------------------------------------\n [INFO] For more information, run Maven with the -e switch \n [INFO] ------------------------------------------------------------------------\n [INFO] Total time: 1 minute 32 seconds\n [INFO] Finished at: Mon Feb 11 22:33:52 CST 2013\n [INFO] Final Memory: 66M/375M\n [INFO] ------------------------------------------------------------------------\n\n$ more /home/xing-yi/gerrit-code/google/gerrit/gerrit-acceptance-tests/target/surefire-reports/com.google.gerrit.acceptance.TestAccount.txt \n-------------------------------------------------------------------------------\nTest set: com.google.gerrit.acceptance.TestAccount\n-------------------------------------------------------------------------------\nTests run: 2, Failures: 0, Errors: 2, Skipped: 0, Time elapsed: 0.061 sec \u003c\u003c\u003c FAILURE!\ninitializationError(com.google.gerrit.acceptance.TestAccount)  Time elapsed: 0.007 sec  \u003c\u003c\u003c ERROR!\njava.lang.Exception: Test class should have exactly one public constructor\n\tat org.junit.runners.BlockJUnit4ClassRunner.validateOnlyOneConstructor(BlockJUnit4ClassRunner.java:136)\n\tat org.junit.runners.BlockJUnit4ClassRunner.validateConstructor(BlockJUnit4ClassRunner.java:125)\n\tat org.junit.runners.BlockJUnit4ClassRunner.collectInitializationErrors(BlockJUnit4ClassRunner.java:114)\n\tat org.junit.runners.ParentRunner.validate(ParentRunner.java:269)\n\tat org.junit.runners.ParentRunner.\u003cinit\u003e(ParentRunner.java:66)\n\tat org.junit.runners.BlockJUnit4ClassRunner.\u003cinit\u003e(BlockJUnit4ClassRunner.java:59)\n\tat org.junit.internal.builders.JUnit4Builder.runnerForClass(JUnit4Builder.java:13)\n\tat org.junit.runners.model.RunnerBuilder.safeRunnerForClass(RunnerBuilder.java:57)\n\tat org.junit.internal.builders.AllDefaultPossibilitiesBuilder.runnerForClass(AllDefaultPossibilitiesBuilder.java:29)\n\tat org.junit.runners.model.RunnerBuilder.safeRunnerForClass(RunnerBuilder.java:57)\n\tat org.junit.internal.requests.ClassRequest.getRunner(ClassRequest.java:24)\n\tat org.apache.maven.surefire.junit4.JUnit4TestSet.\u003cinit\u003e(JUnit4TestSet.java:45)\n\tat org.apache.maven.surefire.junit4.JUnit4DirectoryTestSuite.createTestSet(JUnit4DirectoryTestSuite.java:56)\n\tat org.apache.maven.surefire.suite.AbstractDirectoryTestSuite.locateTestSets(AbstractDirectoryTestSuite.java:96)\n\tat org.apache.maven.surefire.Surefire.createSuiteFromDefinition(Surefire.java:209)\n\tat org.apache.maven.surefire.Surefire.run(Surefire.java:156)\n\tat sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)\n\tat sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:39)\n\tat sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:25)\n\tat java.lang.reflect.Method.invoke(Method.java:597)\n\tat org.apache.maven.surefire.booter.SurefireBooter.runSuitesInProcess(SurefireBooter.java:345)\n\tat org.apache.maven.surefire.booter.SurefireBooter.main(SurefireBooter.java:1009)\n\ninitializationError(com.google.gerrit.acceptance.TestAccount)  Time elapsed: 0.001 sec  \u003c\u003c\u003c ERROR!\njava.lang.Exception: No runnable methods\n\tat org.junit.runners.BlockJUnit4ClassRunner.validateInstanceMethods(BlockJUnit4ClassRunner.java:171)\n\tat org.junit.runners.BlockJUnit4ClassRunner.collectInitializationErrors(BlockJUnit4ClassRunner.java:115)\n\tat org.junit.runners.ParentRunner.validate(ParentRunner.java:269)\n\tat org.junit.runners.ParentRunner.\u003cinit\u003e(ParentRunner.java:66)\n\tat org.junit.runners.BlockJUnit4ClassRunner.\u003cinit\u003e(BlockJUnit4ClassRunner.java:59)\n\tat org.junit.internal.builders.JUnit4Builder.runnerForClass(JUnit4Builder.java:13)\n\tat org.junit.runners.model.RunnerBuilder.safeRunnerForClass(RunnerBuilder.java:57)\n\tat org.junit.internal.builders.AllDefaultPossibilitiesBuilder.runnerForClass(AllDefaultPossibilitiesBuilder.java:29)\n\tat org.junit.runners.model.RunnerBuilder.safeRunnerForClass(RunnerBuilder.java:57)\n\tat org.junit.internal.requests.ClassRequest.getRunner(ClassRequest.java:24)\n\tat org.apache.maven.surefire.junit4.JUnit4TestSet.\u003cinit\u003e(JUnit4TestSet.java:45)\n\tat org.apache.maven.surefire.junit4.JUnit4DirectoryTestSuite.createTestSet(JUnit4DirectoryTestSuite.java:56)\n\tat org.apache.maven.surefire.suite.AbstractDirectoryTestSuite.locateTestSets(AbstractDirectoryTestSuite.java:96)\n\tat org.apache.maven.surefire.Surefire.createSuiteFromDefinition(Surefire.java:209)\n\tat org.apache.maven.surefire.Surefire.run(Surefire.java:156)\n\tat sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)\n\tat sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:39)\n\tat sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:25)\n\tat java.lang.reflect.Method.invoke(Method.java:597)\n\tat org.apache.maven.surefire.booter.SurefireBooter.runSuitesInProcess(SurefireBooter.java:345)\n\tat org.apache.maven.surefire.booter.SurefireBooter.main(SurefireBooter.java:1009)\n\n \n\nI am not sure it is only a my personal case, so I add this patch",
      "parentUuid": "PtqwXY0s",
      "revId": "ff6e8e4a430fad80e7fb632a5424bafb28d415ff",
      "serverId": "173816e5-2b9a-37c3-8a2e-48639d4f1153",
      "unresolved": false
    },
    {
      "key": {
        "uuid": "PtsxBjz4",
        "filename": "/COMMIT_MSG",
        "patchSetId": 4
      },
      "lineNbr": 7,
      "author": {
        "id": 1003873
      },
      "writtenOn": "2013-02-11T15:07:54Z",
      "side": 1,
      "message": "Can you try rebasing on [1] and then reproducing the \u0027mvn test\u0027 error?\n\nWhich maven version are you using?\n\n[1] https://gerrit-review.googlesource.com/42193",
      "parentUuid": "PxhwBVRA",
      "revId": "ff6e8e4a430fad80e7fb632a5424bafb28d415ff",
      "serverId": "173816e5-2b9a-37c3-8a2e-48639d4f1153",
      "unresolved": false
    },
    {
      "key": {
        "uuid": "PxrwYIX8",
        "filename": "/COMMIT_MSG",
        "patchSetId": 4
      },
      "lineNbr": 7,
      "author": {
        "id": 1010015
      },
      "writtenOn": "2013-02-11T16:27:32Z",
      "side": 1,
      "message": "my maven version is Apache Maven 2.2.1 (r801777; 2009-08-07 03:16:01+0800)\nbut the reason is not the maven version , because just had a test again after \u0027rebasing on \n[1]\u0027 and  find [1] can work. I guess  maybe this is the reason that you cannot retrigger this\n issue in your local workspace, however it is impossible for me to find [1] when I created \nthis commit at Feb 9, 2013 5:31 PM, because [1] appears at  Feb 11, 2013 9:23 PM and \nmerged at Feb 11, 2013 11:23 PM. anyway I prefer to discard this commit as I think [1] \nis a better way to fix this issue, thus developer  will be free to give class name.\nonly a bit regret to find [1] has not any background information in its commit message.",
      "parentUuid": "PtsxBjz4",
      "revId": "ff6e8e4a430fad80e7fb632a5424bafb28d415ff",
      "serverId": "173816e5-2b9a-37c3-8a2e-48639d4f1153",
      "unresolved": false
    },
    {
      "key": {
        "uuid": "PtowSp1w",
        "filename": "gerrit-acceptance-tests/src/test/java/com/google/gerrit/acceptance/AccountForTestAcceptance.java",
        "patchSetId": 4
      },
      "lineNbr": 22,
      "author": {
        "id": 1003873
      },
      "writtenOn": "2013-02-11T12:57:33Z",
      "side": 1,
      "message": "This name is ugly.\nHow about UserAccount?",
      "revId": "ff6e8e4a430fad80e7fb632a5424bafb28d415ff",
      "serverId": "173816e5-2b9a-37c3-8a2e-48639d4f1153",
      "unresolved": false
    },
    {
      "key": {
        "uuid": "PxnwOmT8",
        "filename": "gerrit-acceptance-tests/src/test/java/com/google/gerrit/acceptance/AccountForTestAcceptance.java",
        "patchSetId": 4
      },
      "lineNbr": 22,
      "author": {
        "id": 1010015
      },
      "writtenOn": "2013-02-11T14:53:56Z",
      "side": 1,
      "message": "yeah . UserAccount sounds better.",
      "parentUuid": "PtowSp1w",
      "revId": "ff6e8e4a430fad80e7fb632a5424bafb28d415ff",
      "serverId": "173816e5-2b9a-37c3-8a2e-48639d4f1153",
      "unresolved": false
    }
  ]
}